tests/unit/test_ratmama.py

```diff
@@ -12,245 +12,185 @@
+
+
+async def test_announce_from_invalid_user(bot_fx, async_callable_fx, monkeypatch):
+    """
+    Tests that a valid signal received from an invalid user does not trigger a case creation.
+    """
+
+    context = await Context.from_message(bot_fx,
+                                         "#unit_test",
+                                         "some_recruit",
+                                         "Incoming Client: SomeClient - System: Fuelum"
+                                         " - Platform: PC - O2: OK - Language: English (en-US)")
+    monkeypatch.setattr(context, "reply", async_callable_fx)
+
+    await ratmama.handle_ratmama_announcement(context)
+
+    assert not async_callable_fx.was_called
+    assert context.bot.board.find_by_name("SomeClient") is None
+
+
+@pytest.mark.parametrize("signal, platform, system, code_red", [
+    ("ratsignal pc, lhs 3447", Platforms.PC, "LHS 3447", False),
+    ("RATSIGNAL xbox - o2 not ok", Platforms.XB, None, True),
+    ("RatSignal o2 ok;ps4;sol", Platforms.PS, "SOL", False),
+    ("RaTsIgNaL rODentIa    | Pc", Platforms.PC, "RODENTIA", False),
+    ("ratsignal", None, None, False)
+])
+async def test_manual_signal(bot_fx,
+                             async_callable_fx,
+                             monkeypatch,
+                             signal: str,
+                             platform: 'Platform',
+                             system: str,
+                             code_red: bool):
+    """
+    Tests that a manual signal from a standard user is parsed and a case is created accordingly.
+    """
+
+    context = await Context.from_message(bot_fx,
+                                         "#unit_test",
+                                         "some_recruit",
+                                         signal)
+    monkeypatch.setattr(context, "reply", async_callable_fx)
+
+    await ratmama.handle_ratsignal(context)
+
+    rescue = context.bot.board.find_by_name("some_recruit")
+    assert rescue is not None
+    assert rescue.client == "some_recruit"
+    assert rescue.platform == platform
+    assert rescue.system == system
+    assert rescue.code_red == code_red
+
+
+async def test_manual_signal_duplicate(bot_fx, async_callable_fx, monkeypatch):
+    """
+    Tests that a manual signal received for someone who has a case already is replied to as such.
+    """
+
+    context = await Context.from_message(bot_fx,
+                                         "#unit_test",
+                                         "some_recruit",
+                                         "ratsignal Sol, PC, O2 OK")
+    monkeypatch.setattr(context, "reply", async_callable_fx)
+
+    await ratmama.handle_ratsignal(context)
+    await ratmama.handle_ratsignal(context)
+
+    assert async_callable_fx.was_called_with("You already sent a signal, please be patient "
+                                             "while a dispatch is underway.")
```

Comment:
[error] while i recognize this wasn't added by you, can you please adjust this test and the underlying implementation as to match the mecha2 interface while we are here?
```
<unknown> drillsignal ki pc
<Drillsqueak[BOT]> unknown: You already sent a Signal! Please stand by, someone will help you soon!
```